src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIAppWindowCodeFixer.cs

```diff
@@ -0,0 +1,135 @@
+
+            // Register a code action that will invoke the fix.
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    diagnostic.Id,
+                    c => diagnostic.Id == WinUIAppWindowAnalyzer.DiagnosticIdAppWindowNamespace ? FixAppWindowNamespace(context.Document, declaration, apiId!, c)
+                    : diagnostic.Id == WinUIAppWindowAnalyzer.DiagnosticIdAppWindowVarType ? FixAppWindowVarType(context.Document, declaration, c)
+                    : FixAppWindowMember(context.Document, declaration.AncestorsAndSelf().OfType<InvocationExpressionSyntax>().First(), apiId!, varName!, c),
+                    diagnostic.Id + apiId ?? string.Empty),
+                context.Diagnostics);
+        }
+
+        private static async Task<Document> FixAppWindowNamespace(Document document, SyntaxNode invocationExpressionSyntax, string apiId, CancellationToken cancellationToken)
+        {
+            var comment = await CSharpSyntaxTree.ParseText(@$"
+                /*
+                   TODO: Use Microsoft.UI.Windowing.AppWindow for window Management instead of ApplicationView/CoreWindow or Microsoft.UI.Windowing.AppWindow APIs
+                   Read: https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/guides/windowing
+                */
+                ", cancellationToken: cancellationToken).GetRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var oldRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var (newNamespace, newName) = WinUIAppWindowAnalyzer.TypeConversions[apiId].Value;
+            var node = (await CSharpSyntaxTree.ParseText($"{newNamespace}.{newName} x;", cancellationToken: cancellationToken)
+                .GetRootAsync(cancellationToken).ConfigureAwait(false))
+                .DescendantNodesAndSelf().OfType<QualifiedNameSyntax>().First();
+            return document.WithSyntaxRoot(oldRoot!.ReplaceNode(invocationExpressionSyntax, node.WithLeadingTrivia(comment.GetLeadingTrivia())));
+        }
+
+        private static async Task<Document> FixAppWindowVarType(Document document, SyntaxNode invocationExpressionSyntax, CancellationToken cancellationToken)
+        {
+            var documentEditor = await DocumentEditor.CreateAsync(document, cancellationToken).ConfigureAwait(false);
+            VariableDeclarationSyntax variableDeclarator = invocationExpressionSyntax.Ancestors().OfType<VariableDeclarationSyntax>().FirstOrDefault();
+            if (variableDeclarator is not null)
+            {
+                var varIdentifier = variableDeclarator.DescendantNodes().OfType<IdentifierNameSyntax>().Where(id => id.IsVar).FirstOrDefault();
+                if (varIdentifier is not null)
+                {
+                    var node = (await CSharpSyntaxTree.ParseText("Microsoft.UI.Windowing.AppWindow x;", cancellationToken: cancellationToken)
```

Comment:
You may want to take a look at `SyntaxFactory` - it simplifies some of this stuff